contrib/IECoreArnold/src/IECoreArnold/ParameterAlgo.cpp

```diff
@@ -441,6 +441,18 @@ AtArray *dataToArray( const IECore::Data *data, int aiType )
 			return NULL;
 		}
 	}
+	// bools are a special case because of how the STL implements vector<bool>.
+	// since the base for vector<bool> are not actual booleans, we need to manually
+	// convert to an AtArray here.
+	else if( aiType == AI_TYPE_BOOLEAN )
+	{
+		const BoolVectorData* booleans = static_cast<const BoolVectorData *>( data );
```

Comment:
This is the gist of what I was getting at, but let's just make a few small tweaks :

- Call `readable()` once outside the loop rather than twice for every element. The cost is relatively small, but I've measured it in the past and it is worth avoiding.
- Call size() once rather than once per iteration, for the same reason.
- Use `[]` rather than `at()`. We don't need the bounds checking that `at()` will do, since we know we're iterating over the right range. This removes overhead, but it also the intention clearer.
- Use `size_t` (or `vector<bool>::size_type` if you want to out-pedant me) for the loop index.

Here's an example of some of that :

https://github.com/ImageEngine/cortex/blob/master/contrib/IECoreArnold/src/IECoreArnold/MeshAlgo.cpp#L207